### PR TITLE
Make fix for windows, add windows to CI

### PR DESCRIPTION
**Pull request by pamelafox**

### CODE REVIEW

File: src/dotenv_azd/__init__.py [hunk 1]
```diff
@@ -26,17 +27,34 @@ class AzdNoProjectExistsError(AzdError):
 
 def _azd_env_get_values(cwd: str | bytes | PathLike | None = None) -> str:
     try:
-        result = run(["/usr/bin/env", "azd", "env", "get-values"], capture_output=True, text=True, cwd=cwd, check=True)
+        # Try running azd directly (cross-platform approach)
+        result = run(["azd", "env", "get-values"], capture_output=True, text=True, cwd=cwd, check=True)
+    except FileNotFoundError:
+        # If direct command fails, try finding azd in path
+        azd_path = shutil.which("azd")
```
Comment (cedricvidal) [inline src/dotenv_azd/__init__.py]: Oh, is it possible to get a `FileNotFoundError` and yet find `azd` in the `PATH` using `which`? I would expect the `subprocess.run` command to use the same executable search logic as `which` somehow.

Is this specific to Windows?

File: tests/test_load_azd_env.py [hunk 2]
```diff
@@ -79,3 +76,48 @@ def test_load_azd_env_azd_command_not_found_error(tmp_path: Path) -> None:
 
 def test_load_azd_env_ignore_errors(tmp_path: Path) -> None:
     load_azd_env(cwd=tmp_path, quiet=True)
+
+
+@patch("dotenv_azd.run")
```
Comment (cedricvidal) [inline tests/test_load_azd_env.py]: Great test coverage of all corner cases :)

File: src/dotenv_azd/__init__.py [hunk 3]
```diff
@@ -26,17 +27,34 @@ class AzdNoProjectExistsError(AzdError):
...
+        if not azd_path:
+            msg = "azd command not found, install it prior to using dotenv-azd"
+            raise AzdCommandNotFoundError(msg)
+        try:
+            result = run([azd_path, "env", "get-values"], capture_output=True, text=True, cwd=cwd, check=True)
+        except CalledProcessError as e:
```
Comment (cedricvidal) [inline src/dotenv_azd/__init__.py]: not blocking but if you feel like it: code path redunduncies, maybe extract the command running and `which` fallback handling to a separate method to keep the error handling logic centralized?

File: src/dotenv_azd/__init__.py (hunk 1 above)
Comment (pamelafox) [reply src/dotenv_azd/__init__.py]: I don't actually know if which could find it in some cases, this logic was suggested by Copilot. I don't have a Windows setup right now, so that's why I added windows to the CI matrix. I know that "azd" without "usr/bin" does work on Windows, as that's what Abril used, so we could remove this for now.

File: src/dotenv_azd/__init__.py (hunk 3 above)
Comment (pamelafox) [reply src/dotenv_azd/__init__.py]: lol yeah Copilot wasnt very DRY here! Removing since I'm removing which path anyway

File: tests/test_load_azd_env.py [hunk 4]
```diff
@@ -29,53 +30,55 @@ def _azd_env_set(key: str, value: str, *, cwd: Path) -> str:
...
 
-def test_load_azd_env_override(tmp_path: Path) -> None:
-    from os import environ, getenv
+def test_load_azd_env_override(tmp_path: Path, monkeypatch) -> None:
 
     with open(tmp_path / "azure.yaml", "w") as config:
         config.write("name: dotenv-azd-test\n")
 
-    environ["VAR1"] = "INITIAL"
+    monkeypatch.setenv("VAR1", "INITIAL")
```
Comment (pamelafox) [inline tests/test_load_azd_env.py:48]: Safer to use monkeypatch.setenv in tests

File: tests/test_load_azd_env.py [hunk 5]
```diff
@@ -29,53 +30,55 @@ def _azd_env_set(key: str, value: str, *, cwd: Path) -> str:
...
     _azd_env_new("MY_AZD_ENV", cwd=tmp_path)
     _azd_env_set("VAR1", "OVERRIDE", cwd=tmp_path)
     var_set = load_azd_env(cwd=tmp_path)
-    assert getenv("AZURE_ENV_NAME") == "MY_AZD_ENV"
-    assert getenv("VAR1") == "INITIAL"
+    assert os.getenv("AZURE_ENV_NAME") == "MY_AZD_ENV"
```
Comment (pamelafox) [inline tests/test_load_azd_env.py:52]: Just changed to os.getenv to make it clear its coming from os

**Comment by pamelafox** [comment]
Simplifications made!

**Comment by pamelafox** [comment]
Please run the workflow and see if it succeeds on windows

File: tests/test_load_azd_env.py [hunk 6]
```diff
@@ -29,53 +30,55 @@ def _azd_env_set(key: str, value: str, *, cwd: Path) -> str:
...
     with open(tmp_path / "azure.yaml", "w") as config:
         config.write("name: dotenv-azd-test\n")
 
-    environ["VAR1"] = "INITIAL"
+    monkeypatch.setenv("VAR1", "INITIAL")
     _azd_env_new("MY_AZD_ENV", cwd=tmp_path)
     _azd_env_set("VAR1", "OVERRIDE", cwd=tmp_path)
     var_set = load_azd_env(cwd=tmp_path)
-    assert getenv("AZURE_ENV_NAME") == "MY_AZD_ENV"
-    assert getenv("VAR1") == "INITIAL"
```
Comment (cedricvidal) [inline tests/test_load_azd_env.py:54]: It seems to me `assert getenv("VAR1") == "INITIAL"` is still wanted to test the non override mode, no?

File: tests/test_load_azd_env.py (hunk 5 above)
Comment (cedricvidal) [reply tests/test_load_azd_env.py:52]: Isn't it better to narrow down what's imported? but maybe it's not as important as for Typescript, I'm not so intimate with the subtelties in Python of importing a module versus an exported object of the module

File: tests/test_load_azd_env.py (hunk 4 above)
Comment (cedricvidal) [reply tests/test_load_azd_env.py:48]: Sounds good

File: tests/test_load_azd_env.py [hunk 7]
```diff
@@ -29,53 +30,55 @@ def _azd_env_set(key: str, value: str, *, cwd: Path) -> str:
...
-
-    from dotenv_azd import AzdCommandNotFoundError
-
-    path = environ["PATH"]
-    environ["PATH"] = ""
+def test_load_azd_env_azd_command_not_found_error(tmp_path: Path, monkeypatch) -> None:
+    monkeypatch.setenv("PATH", "")
     with pytest.raises(AzdCommandNotFoundError):
         load_azd_env(cwd=tmp_path)
-    environ["PATH"] = path
```
Comment (cedricvidal) [inline tests/test_load_azd_env.py:77]: it's not needed to restore the PATH before the method exits? is `monkeypatch.setenv` taking care of this?

File: tests/test_load_azd_env.py [hunk 8]
```diff
@@ -29,53 +30,55 @@ def _azd_env_set(key: str, value: str, *, cwd: Path) -> str:
...
-
 
 def test_load_azd_env_ignore_errors(tmp_path: Path) -> None:
     load_azd_env(cwd=tmp_path, quiet=True)
+
+@patch("dotenv_azd.run")
+def test_cross_platform_direct_call_succeeds(mock_run):
+    """Test that the function works when the direct call to azd succeeds."""
+    # Mock subprocess.run to return success
+    mock_run.return_value = subprocess.CompletedProcess(
```
Comment (cedricvidal) [inline tests/test_load_azd_env.py:77]: very nice way of providing output from a sub process!

File: src/dotenv_azd/__init__.py [hunk 9]
```diff
@@ -1,5 +1,6 @@
 from __future__ import annotations
 
+import shutil
```
Comment (cedricvidal) [inline src/dotenv_azd/__init__.py]: seems this import is not necessary anymore

File: tests/test_load_azd_env.py (hunk 6 above)
Comment (cedricvidal) [reply tests/test_load_azd_env.py:54]: Nevermind, the check is bellow

**Comment by cedricvidal** [comment]
@pamelafox  Suggestion, add the classifiers for Windows, MacOS and Linux in `pyproject.toml`

```
  "Operating System :: Microsoft :: Windows",
  "Operating System :: POSIX :: Linux",
  "Operating System :: MacOS :: MacOS X",
```

Source:
https://pypi.org/classifiers/

File: tests/test_load_azd_env.py (hunk 5 above)
Comment (pamelafox) [reply tests/test_load_azd_env.py:52]: I don't think we get the same benefits as in TypeScript, though its a good question. I like the readability of seeing where a function came from, and for os, I feel like we always "import os". I'll ask around.
Functionality wise, nothing should be affected.

File: tests/test_load_azd_env.py (hunk 7 above)
Comment (pamelafox) [reply tests/test_load_azd_env.py:77]: Correct, monkeypatch takes care of it

Comment (cedricvidal) [reply tests/test_load_azd_env.py:77]: Sounds good

File: tests/test_load_azd_env.py (hunk 5 above)
Comment (cedricvidal) [reply tests/test_load_azd_env.py:52]: Ok

**Review by cedricvidal — APPROVED** [review]
